### PR TITLE
first pass at best-practice requires

### PR DESCRIPTION
**Pull request by geemus**
Here is at least a first pass at this, may be some gotchas/issues remaining. The hope here is that it moves toward gem best practices and reduces how many `/lib/fog` instances we have on the load_path, which should help with performance (and possibly some weird edge-case load stuff).

In particular I didn't leave behind a way to still have `require "fog/brightbox"` to work. Trying to decide how best to approach, leaving `lib/fog/brightbox` in here just for that defeats some of the purpose. I was contemplating having it just be a shim (perhaps with deprecation warning) inside fog-core or similar.

I think I'll probably leave fog-core dirs as they are, instead of moving toward `lib/fog-core` or something, and this would give a consistent place to do that from I think/hope.

Anyway, certainly welcome feedback, looked like my local tests are passing fine.

One question, seems like there are parallel nestings of specs? some in compute/storage, some more deeply nested. Is one or the other preferred and/or is there a reason for the distinction? I thought about remapping spec dirs to reflect the lib dir movement, but this part left me a bit puzzled.

/cc @tokengeek @plribeiro3000

### CODE REVIEW

File: lib/fog/brightbox.rb [hunk 1]
```diff
@@ -1,46 +0,0 @@
-require "fog/core"
-require "fog/json"
-
-# See the main fog gem for more details about the top level class {Fog}
```
Comment (plribeiro3000) [inline lib/fog/brightbox.rb]: Just wondering. Wont this break?
AFAIK bundler will try to load the path `fog/brightbox` when this gem is listed in a `Gemfile` and thus will raise an exception.

**Comment by geemus** [comment]
Was just thinking about this again, obviously this no longer will work, but I'd still like to do something in this vein if possible to see about easing load-path pains. Maybe it is a mute point if we split out all the gems though.

**Comment by plribeiro3000** [comment]
Agree. I think that once we split all providers it will get better. When i
got back from my vacation i will star extrqcting again.

Em ter, 5 de jan de 2016 13:17, Wesley Beary <notifications@github.com>
escreveu:

**Comment by geemus** [comment]
Yeah, I was also thinking I would try to find time to help with further extractions. I guess I just wondered if we could do some of this style of restructuring along with the extractions. Sort of two-bird-with-one-stone. That said, doing two kinds of changes certainly introduces additional risk, so maybe best to do them sequentially anyway.

@plribeiro3000 enjoy your vacation!

**Comment by tokengeek** [comment]
@geemus I'll close this up then.

I've not had much time for anything so unable to help at the moment.

**Comment by plribeiro3000** [comment]
Thanks @geemus. I was thinking about The sequenti solution. Every little
step we dos broke something só i figured we ahould go step by step but
given The time to do so i don't know if that exactly The best path.

Em ter, 5 de jan de 2016 14:42, Paul Thornthwaite <notifications@github.com>
escreveu:

**Comment by geemus** [comment]
Yeah, sounds good. Thanks!

**Pull request closed without merging**